### PR TITLE
feat: allow mine nft creation transaction

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

File: txstratum/api.py [hunk 1]
```diff
@@ -101,11 +101,13 @@ async def submit_job(self, request: web.Request) -> web.Response:
...
             return web.json_response({'error': 'tx-weight-is-too-high'}, status=400)
 
+        is_nft_creation = tx.is_nft_creation
+
         for txout in tx.outputs:
             if len(txout.script) > self.max_output_script_size:
                 self.log.debug('txout-script-is-too-big', data=data)
                 return web.json_response({'error': 'txout-script-is-too-big'}, status=400)
-            if self.only_standard_script:
+            if self.only_standard_script and not is_nft_creation:
```
Comment (msbrogli) [inline txstratum/api.py:110]: This code is outdated, isn't it?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: txstratum/api.py (hunk 1 above)
Comment (pedroferreira1) [reply txstratum/api.py]: Yes, can you see again please?

File: tests/test_api.py [hunk 2]
```diff
@@ -208,7 +221,7 @@ async def test_submit_job_invalid_output_script_size(self):
         resp = await self.client.request('POST', '/submit-job', json={'tx': tx_hex})
         data = await resp.json()
         self.assertEqual(400, resp.status)
-        self.assertEqual({'error': 'txout-script-is-too-big'}, data)
+        self.assertEqual({'error': 'tx-non-standard'}, data)
```
Comment (msbrogli) [inline tests/test_api.py:224]: I'd call it `non-standard-tx`.

File: tests/test_api.py [hunk 3]
```diff
@@ -220,7 +233,7 @@ async def test_submit_job_invalid_non_standard_script(self):
         resp = await self.client.request('POST', '/submit-job', json={'tx': tx_hex})
         data = await resp.json()
         self.assertEqual(400, resp.status)
-        self.assertEqual({'error': 'txout-non-standard-script'}, data)
+        self.assertEqual({'error': 'tx-non-standard'}, data)
```
Comment (msbrogli) [inline tests/test_api.py:236]: I'd call it `non-standard-tx`.

File: txstratum/api.py [hunk 4]
```diff
@@ -101,14 +100,9 @@ async def submit_job(self, request: web.Request) -> web.Response:
...
-            if len(txout.script) > self.max_output_script_size:
-                self.log.debug('txout-script-is-too-big', data=data)
-                return web.json_response({'error': 'txout-script-is-too-big'}, status=400)
-            if self.only_standard_script:
-                p2pkh = P2PKH.parse_script(txout.script)
-                if p2pkh is None:
-                    return web.json_response({'error': 'txout-non-standard-script'}, status=400)
+        if not tx.is_standard(self.max_output_script_size, self.only_standard_script):
+            self.log.debug('tx-non-standard', data=data)
+            return web.json_response({'error': 'tx-non-standard'}, status=400)
```
Comment (msbrogli) [inline txstratum/api.py:105]: I'd call it `non-standard-tx`.

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

File: tests/test_api.py (hunk 2 above)
Comment (pedroferreira1) [reply tests/test_api.py]: Done

File: tests/test_api.py (hunk 3 above)
Comment (pedroferreira1) [reply tests/test_api.py]: Done

File: txstratum/api.py (hunk 4 above)
Comment (pedroferreira1) [reply txstratum/api.py]: Done

**Pull request merged**